tests/trading/test_wcl_fuzzy.py

```diff
@@ -89,33 +94,35 @@ def acquireShortShareSet(outcome, amount, approvalAddress, sender):
     assert order[SHARES_ESCROWED] == creatorLongShares or creatorShortShares
 
     # fill order
+    fillerEthRequiredLong = 0 if fillerLongShares == 0 else fillerLongShares * market.getNumTicks()
+    fillerEthRequiredShort = 0 if fillerShortShares == 0 else fillerShortShares * market.getNumTicks()
     acquireLongShares(orderOutcome, fillerLongShares, fillOrder.address, sender = fillerKey)
     acquireShortShareSet(orderOutcome, fillerShortShares, fillOrder.address, sender = fillerKey)
     remaining = fillOrder.publicFillOrder(orderID, orderSize, 42, sender = fillerKey, value = fillerTokens)
     assert not remaining
 
     # assert final state
-    assert cash.balanceOf(creatorAddress) == expectedMakerTokens
-    assert cash.balanceOf(fillerAddress) == expectedFillerTokens
+    assert fixture.chain.head_state.get_balance(creatorAddress) == creatorOriginalBalance - creatorEthRequiredLong - creatorEthRequiredShort - creatorTokens + expectedMakerTokens
+    assert fixture.chain.head_state.get_balance(fillerAddress) == fillerOriginalBalance - fillerEthRequiredLong - fillerEthRequiredShort - fillerTokens + expectedFillerTokens
```

Comment:
This change feels incorrect.  We should probably rename `expectedMakerTokens` and `expectedFillerTokens` to `expectedMakerEth` and `expectedTakerEth` but it seems like we should still be testing against the incoming expectation directly.

If this change is necessary, can you explain a bit why that is so?

Reply:
I agree that `expectedMakerTokens` and `expectedFillerTokens` should be renamed.  However, comparing the ETH balance of the creator/filler to these values in the same way we were dong with the cash balance will fail.  The ETH balance of the creator is set very high at the beginning of the test, whereas the `expectedMakerTokens` and `expectedFillerTokens` are typically set to values like `shortCost`, `shortCost - shortFee`, 0, etc.  

I agree that it seems odd to do the assert in this way, but in order to do it like we were before, I think we'd have to alter the values we're passing in for `expectedMakerTokens` and `expectedFillerTokens`.

Reply:
Do all of the test accounts have a bunch of ETH in them, or only a the first one?  One option would be to transfer away the excess ETH before the tests start, so the balance starts at "exactly the right amount" and ends at "exactly the right amount".  Even as it is, I would expect `creatorOriginalBalance + expectedMakerTokens` and `fillerOriginalBalance + expectedFillerTokens`.  I would not expect all of the stuff in the middle, as that suggests there is a bug in the test somewhere.

Reply:
What if we did asserts on the balance of the creator/filler after the creator acquires shares, after the order is created, after the filler acquires shares, and after the order is filled?  That way, the final asserts would only be comparing the users' current balances with their previous balance + the expected maker/filler tokens?  Plus, it would verify their balances are correct along the way.